src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -484,4 +486,17 @@ public void close() throws IOException {
             // Else retry in case a new buf was CASed in fill()
         }
     }
+
+    @Override
+    public synchronized long transferTo(OutputStream out) throws IOException {
+        Objects.requireNonNull(out, "out");
+        int avail = count - pos;
+        if (avail > 0) {
+            out.write(buf, pos, avail);
```

Comment:
I think `buf` should not be accessed directly but via `getBufIfOpen()` because we need to throw IOE in case `this` is closed.